contrib/golang/filters/http/source/go/pkg/http/type.go

```diff
@@ -137,11 +129,134 @@ func (h *httpHeaderMap) Range(f func(key, value string) bool) {
 	}
 }
 
-// ByteSize return size of HeaderMap
-func (h *httpHeaderMap) ByteSize() uint64 {
-	return h.headerBytes
+// api.RequestHeaderMap
+type requestHeaderMapImpl struct {
+	requestOrResponseHeaderMapImpl
+}
+
+var _ api.RequestHeaderMap = (*requestHeaderMapImpl)(nil)
+
+func (h *requestHeaderMapImpl) Protocol() string {
+	v, _ := h.Get(":protocol")
+	return v
+}
+
+func (h *requestHeaderMapImpl) Scheme() string {
+	v, _ := h.Get(":scheme")
+	return v
+}
+
+func (h *requestHeaderMapImpl) Method() string {
+	v, _ := h.Get(":method")
+	return v
+}
+
+func (h *requestHeaderMapImpl) Path() string {
+	v, _ := h.Get(":path")
+	return v
+}
+
+func (h *requestHeaderMapImpl) Host() string {
+	v, _ := h.Get(":authority")
+	return v
+}
+
+// api.ResponseHeaderMap
+type responseHeaderMapImpl struct {
+	requestOrResponseHeaderMapImpl
+}
+
+var _ api.ResponseHeaderMap = (*responseHeaderMapImpl)(nil)
+
+func (h *responseHeaderMapImpl) Status() int {
+	if str, ok := h.Get(":status"); ok {
+		v, _ := strconv.Atoi(str)
+		return v
+	}
+	return 0
```

Comment:
0 means get status fail? It also has another meaning similar to the 499 code. such as below: https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#format-strings
```
HTTP response code. Note that a response code of ‘0’ means that the server never sent the beginning of a response. This generally means that the (downstream) client disconnected.
```

Reply:
Yes, 0 means not found status in response, I see envoy Internal implementation also seem to take 0 as the default:

* https://github.com/envoyproxy/envoy/blob/main/source/common/http/utility.cc#L593
* https://github.com/envoyproxy/envoy/blob/main/source/common/formatter/substitution_formatter.cc#L1026

Do you think it's better to define it as `Status() (int, bool)`, if get fail, `return 0, false`?

Reply:
yeah